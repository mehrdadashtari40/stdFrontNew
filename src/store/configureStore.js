import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
