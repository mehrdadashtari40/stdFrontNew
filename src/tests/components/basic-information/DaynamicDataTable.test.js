import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../store/configureStore';
import DaynamicDataTable from '../../../views/basic-information/components/DynamicDataTable';

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , DaynamicDataTable Component', () => {
	const wrapper = shallow(
		<DaynamicDataTable
			structure={['test1 : testValue1', 'test2 : testValue2', 'test3 : testValue3', 'test4 : testValue4']}
			data={['test : testValue', 'id : 11111']}
		/>
	);

	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(DaynamicDataTable.prototype, 'componentDidMount');
	shallow(
		<DaynamicDataTable
			structure={['test1 : testValue1', 'test2 : testValue2', 'test3 : testValue3', 'test4 : testValue4']}
			data={['test : testValue', 'id : 11111']}
		/>
	);

	expect(DaynamicDataTable.prototype.componentDidMount).toHaveBeenCalled();
	DaynamicDataTable.prototype.componentDidMount.mockRestore();
});

it('invokes `componentWillMount` when mounted', () => {
	jest.spyOn(DaynamicDataTable.prototype, 'componentWillMount');
	shallow(
		<DaynamicDataTable
			structure={['test1 : testValue1', 'test2 : testValue2', 'test3 : testValue3', 'test4 : testValue4']}
			data={['test : testValue', 'id : 11111']}
		/>
	);

	expect(DaynamicDataTable.prototype.componentWillMount).toHaveBeenCalled();
	DaynamicDataTable.prototype.componentWillMount.mockRestore();
});


it('invokes `shouldComponentUpdate` when mounted', () => {
	jest.spyOn(DaynamicDataTable.prototype, 'shouldComponentUpdate');
	shallow(
		<DaynamicDataTable
			structure={['test1 : testValue1', 'test2 : testValue2', 'test3 : testValue3', 'test4 : testValue4']}
			data={['test : testValue', 'id : 11111']}
		/>
	);

	expect(DaynamicDataTable.prototype.shouldComponentUpdate).toHaveBeenCalled();
	DaynamicDataTable.prototype.shouldComponentUpdate.mockRestore();
});
