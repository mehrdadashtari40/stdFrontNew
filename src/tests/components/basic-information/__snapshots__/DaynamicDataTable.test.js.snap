// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , DaynamicDataTable Component 1`] = `
<div>
  <Datatable
    className="table table-striped table-bordered table-hover"
    filter={true}
    options={
      Object {
        "buttons": false,
        "columns": Array [
          Object {
            "class": "",
            "data": null,
            "render": [Function],
          },
          Object {
            "class": "",
            "data": null,
            "render": [Function],
          },
          Object {
            "class": "",
            "data": null,
            "render": [Function],
          },
          Object {
            "class": "",
            "data": null,
            "render": [Function],
          },
          Object {
            "class": "",
            "data": null,
            "render": [Function],
          },
        ],
        "data": Array [
          "test : testValue",
          "id : 11111",
        ],
        "iDisplayLength": 10,
        "order": Array [],
        "rowID": "",
      }
    }
    paginationLength={true}
    width="100%"
  >
    <thead>
      <tr>
        <th
          className="hasinput"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          <input
            className="form-control "
            id="dateselect_filter"
            type="text"
          />
        </th>
        <th
          className="hasinput"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          <input
            className="form-control "
            id="dateselect_filter"
            type="text"
          />
        </th>
        <th
          className="hasinput"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          <input
            className="form-control "
            id="dateselect_filter"
            type="text"
          />
        </th>
        <th
          className="hasinput"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          <input
            className="form-control "
            id="dateselect_filter"
            type="text"
          />
        </th>
        <th
          style={
            Object {
              "width": "3%",
            }
          }
        />
      </tr>
      <tr>
        <th
          data-hide="phone"
        >
          <i
            className=""
          />
        </th>
        <th
          data-hide="phone"
        >
          <i
            className=""
          />
        </th>
        <th
          data-hide="phone"
        >
          <i
            className=""
          />
        </th>
        <th
          data-hide="phone"
        >
          <i
            className=""
          />
        </th>
        <th
          style={
            Object {
              "width": "3%",
            }
          }
        >
          عملیات
        </th>
      </tr>
    </thead>
  </Datatable>
  <div
    aria-hidden="true"
    aria-labelledby="ServiceModalLabel"
    className="modal fade"
    id="editRowModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      style={
        Object {
          "width": "65%",
        }
      }
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-hidden="true"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
            id="ServiceModalLabel"
            style={
              Object {
                "display": "inline-block",
              }
            }
          />
        </div>
        <div
          className=""
        >
          <ModalForm
            data={Array []}
            edit={false}
            structure={
              Array [
                "test1 : testValue1",
                "test2 : testValue2",
                "test3 : testValue3",
                "test4 : testValue4",
              ]
            }
            submit={[Function]}
            t={[Function]}
          />
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn_inbox btn_datatable"
            data-dismiss="modal"
            type="button"
          >
            <i
              className="fa fa-window-close "
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
